### ISSUE TITLE
Presence not sent out (in cases where it should) after page refresh

### ISSUE THREAD

**Issue by grep-i** [jcbrand/converse.js#536]
While using version 0.10.0, after refresh page all incoming messages go into offline storage and webclient can't send messages. Only after change status( for example online -> busy) it works normal.
There is no problem with previous versions. 
Chat using as plugin of roundcube.

**Comment by pmkaboo** [comment]
ive run into the same problem, tracked it down to this part of [converse.js](https://github.com/jcbrand/converse.js/blob/master/converse.js#L4942):
```javascript
            fetch: function () {
                this.model.fetch({
                    silent: true, // We use the success handler to handle groups that were added,
                                  // we need to first have all groups before positionFetchedGroups
                                  // will work properly.
                    success: function (collection, resp, options) {
                        if (collection.length !== 0) {
                            this.positionFetchedGroups(collection, resp, options);
                        }
                        converse.roster.fetch({
                            add: true,
                            success: function (collection) {
                                if (collection.length > 0) {
                                    converse.initial_presence_sent = 1;
                                } else {
                                    // We don't have any roster contacts stored
                                    // in sessionStorage, so lets fetch the
                                    // roster from the XMPP server.
                                    converse.roster.fetchFromServer();
                                }
                            }
                        });
                    }.bind(this)
                });
                return this;
            },
```

if i understand it correctly, it sets `initial_presence_sent` to 1 when it receives roster from session storage, this will ofcourse happen after page refresh and thus the presence stanza wont be sent and youll be unable to receive any messages. changing status is one way to bypass this.

**Comment by jcbrand** [comment]
> this will of course happen after page refresh and thus the presence stanza wont be sent and youll be unable to receive any messages

If you still have the same BOSH session after a page refresh (which would normally be the case), then you should still receive messages without having to broadcast your presence again.

However, if you are creating a new BOSH session for each page load, then the above problem might occur.

@pmkaboo Are you also using the roundcube plugin?

I can perhaps add a config setting to force the presence to be sent on each page load. However, you might first want to check that you're not unnecessarily setting up a new BOSH session for each page load.

**Comment by pmkaboo** [comment]
not using roundcube, and i forgot to mention that im connecting through websockets.

**Comment by jcbrand** [comment]
@pmkaboo Ok, using websockets could very well be causing this problem because currently there isn't yet a way to maintain a single session across page loads, so a new session is created each time. I guess for now, a simple fix would be to just check whether websockets is being used when deciding whether to send the presence upon page load.

**Comment by grep-i** [comment]
Could you add a config setting to force the presence to be sent on each page load?
Converse in roundcube-plugin keep session in Roundcube session's storage, using converse.min.js with converse-rcmail.js. It seems, this mechanism are broken after updating to version 0.10. Unfortunately, i have no skill in js and difficult understand reason of trouble.

**Comment by jcbrand** [comment]
@grep-i I need to do a little research first, but I'll likely add such a config option. However that doesn't mean it'll magically work in the Roundcube plugin, that plugin will likely also have to be updated. You'll have to ask @priyadi for that.

**Comment by irek-zaripov** [comment]
Adding the sting https://github.com/irek-zaripov/converse.js/blob/master/converse.js#L4942 
will send presence status every time after fetching roster from the SessionStorage. I have tested it in in the roundcube plugin.

**Comment by irek-zaripov** [comment]
But it seems there is a new issue. The roster contacts statuses are not refreshed from the XMPP server after fetching from the SessionStorage.

**Comment by jcbrand** [comment]
@pmkaboo, @irek-zaripov, @grep-i This issue should hopefully be fixed now. Can you please give it a test and confirm whether it's working for you?

**Comment by grep-i** [comment]
@jcbrand It works fine, thanks a lot.
